labeling/internal/java.go

```diff
@@ -0,0 +1,23 @@
+package internal
+
+import (
+	"github.com/CircleCI-Public/circleci-config/labeling/codebase"
+	"github.com/CircleCI-Public/circleci-config/labeling/labels"
+	"path"
+)
+
+var JavaRules = []labels.Rule{
+	func(c codebase.Codebase, ls *labels.LabelSet) (label labels.Label, err error) {
+		label.Key = labels.DepsJava
+		pomXml, err := c.FindFile("pom.xml", "gradlew")
+		label.Valid = pomXml != ""
+		label.BasePath = path.Dir(pomXml)
+		return label, err
+	},
+	func(c codebase.Codebase, ls *labels.LabelSet) (label labels.Label, err error) {
+		label.Key = labels.ToolGradle
+		gradlew, err := c.FindFile("gradlew")
+		label.Valid = gradlew != "" && path.Dir(gradlew) == (*ls)[labels.DepsJava].BasePath
```

Comment:
~ I think in hindsight the rules functions should not use pointers to the labelset, since maps are already pointers, so we wouldn't need the ugly (*ls) here.  I can make a PR to switch that over if others agree.

Reply:
Sounds good to me

Reply:
https://github.com/CircleCI-Public/circleci-config/pull/51